src/context/WatchLaterProvider.jsx

```diff
@@ -0,0 +1,85 @@
+import axios from "axios";
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { useAuth } from "./AuthProvider";
+
+const WatchLaterContext = createContext();
+
+export default function WatchLaterProvider({ children }) {
+
+    const [watchLaterVideos,setWatchLaterVideos] = useState([]);
+    const { auth} = useAuth();
+
+    const addWatchLaterVideo = (video,setLoader=()=>{}) => {
+        axios.request({
+            method: "post",
+            url: "/api/user/watchlater",
+            headers: {authorization: auth.authToken},
+            data:{
+                video
+            }
+        }).then((res)=>{
+            setWatchLaterVideos([...res.data.watchlater])
+            setLoader(false);
+        }).catch((err)=>{
+            console.log(err)
```

Comment:
1 console log can be removed.

Reply:
that I have kept for now to detect error if any. Later I will replace it with toast.